Expose rz_core_prevop_addr* as RZ_API + docs

thestr4ng3r opened the pull request:
<!-- Filling this template is mandatory -->

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

This was sort of a regression from
59b38e6efaf00b9b9869e0ec5baba4f1b9605f37, where it was switched to RZ_IPI in the header to be consistent with the source. However we need RZ_API because it is used at least in Cutter.

**Test plan**

Build cutter dev against this. This should not happen:
```
Undefined symbols for architecture arm64:
  "_rz_core_prevop_addr", referenced from:
      XrefsDialog::updatePreview(unsigned long long)::$_2::operator()(rz_core_t*) const in XrefsDialog.cpp.o
  "_rz_core_prevop_addr_force", referenced from:
      CutterCore::prevOpAddr(unsigned long long, int) in Cutter.cpp.o
      XrefsDialog::updatePreview(unsigned long long)::$_2::operator()(rz_core_t*) const in XrefsDialog.cpp.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```

thestr4ng3r committed:
Expose rz_core_prevop_addr* as RZ_API + docs

This was sort of a regression from
59b38e6efaf00b9b9869e0ec5baba4f1b9605f37, where it was switched to
RZ_IPI in the header to be consistent with the source. However we need
RZ_API because it is used at least in Cutter.

XVilka reviewed (changes requested):
Can you move it out of TUI then? Maybe in `librz/core/canalysis.c` of something similar

thestr4ng3r committed:
Move rz_core_prevop_addr* to canalysis.c

Since it is api, it is not bound to TUI only, so fits better in generic
core analysis code.

thestr4ng3r commented:
Agreed, moved.

wargio commented on librz/include/rz_core.h:
```diff
@@ -484,8 +484,8 @@ RZ_API RzLineNSCompletionResult *rz_core_autocomplete_rzshell(RzCore *core, RzLi
 RZ_IPI void rz_core_print_scrollbar(RzCore *core);
 RZ_IPI void rz_core_print_scrollbar_bottom(RzCore *core);
```
move these IPI within the private header

thestr4ng3r replied:
That is unrelated, better make another pr for it.

wargio commented on librz/core/canalysis.c:
```diff
@@ -6331,3 +6331,106 @@ RZ_API void rz_core_analysis_calls(RZ_NONNULL RzCore *core, bool imports_only) {
+ *
+ * \return if analysis was able to find the previous instruction address
+ */
+RZ_API bool rz_core_prevop_addr(RzCore *core, ut64 start_addr, int numinstrs, ut64 *prev_addr) {
+	RzAnalysisBlock *bb;
```
Add ownership and check if pointers are NULL or not.

wargio commented on librz/core/canalysis.c:
```diff
@@ -6331,3 +6331,106 @@ RZ_API void rz_core_analysis_calls(RZ_NONNULL RzCore *core, bool imports_only) {
+ * Like rz_core_prevop_addr(), but also uses heuristics as fallback if
+ * no concrete analysis info is available.
+ */
+RZ_API ut64 rz_core_prevop_addr_force(RzCore *core, ut64 start_addr, int numinstrs) {
+	int i;
+	for (i = 0; i < numinstrs; i++) {
+		start_addr = rz_core_prevop_addr_heuristic(core, start_addr);
```
move `int i` inside the for, check if core is NULL and add ownership

thestr4ng3r committed:
Add asserts and annotations

thestr4ng3r committed:
Merge branch 'dev' into prevop-api
